projects/go-lib/src/lib/components/go-layout/go-layout.component.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  Component,
+  OnInit,
+  ViewEncapsulation
+} from '@angular/core';
+
+import {
+  ActivatedRoute,
+  NavigationCancel,
+  NavigationEnd,
+  NavigationError,
+  NavigationStart,
+  Router,
+  RouterOutlet
+} from '@angular/router';
+
+import { routerAnimation } from '../../animations/route.animation';
+import { fadeTemplateAnimation } from '../../animations/fade.animation';
+
+@Component({
+  selector: 'go-layout',
+  templateUrl: './go-layout.component.html',
+  styleUrls: ['./go-layout.component.scss'],
+  animations: [routerAnimation, fadeTemplateAnimation],
+  encapsulation: ViewEncapsulation.None
+})
+export class GoLayoutComponent implements OnInit {
+
+  routeLoader: boolean = false;
+
+  constructor(private router: Router) { }
+
+  ngOnInit(): void {
+    this.router.events.subscribe(event => {
+      if (event instanceof NavigationStart) {
+        this.routeLoader = true;
+      }
+      if ((event instanceof NavigationEnd) || (event instanceof NavigationCancel) || (event instanceof NavigationError)) {
```

Comment:
Should this be an else if? It shouldn't matter from a programming point of view as they don't overlap at all, but it does read a little easier for someone coming along later.

Reply:
@StevenUlmer I suppose that's fair. I changed it to `else if`. I only did it that way because I felt like it was two separate bits of functionality. One was showing the loader, one was hiding, but it doesn't matter either way and `else if` probably is easier.